### ISSUE TITLE
Poor performance with Redis adapter

### ISSUE THREAD

**Issue by fibis** [soketi/soketi#397]
We have deployed the latest soketi version via the helm chart into our Kubernetes cluster, to replace the Laravel echo server in our application.

We had used Redis as adapter driver with a managed Redis instance on gcp. Since soketi should be really fast, we were confused that the CPU usage of the pods was really bad. With the autoscaler applied and adjusted cpu limits of 2 vCPUs per pod, the autoscaler calls up to 20 pods to manage the load. Also, the pods where OOMKilled the whole time. That was unsatisfying.

Then we switched to the cluster adapter and now, we just have 3 pods (for HA) running with de default resource constraints from the helm chart. Also, there were no pod restarts.

Was this a problem of the Redis adapter in general, or maybe a problem of the managed Redis database?

Also, after switching to the cluster adapter, the metrics are more accurate:
![image](https://user-images.githubusercontent.com/7265809/156762416-9c67fc41-2134-4048-9e6d-4401d5ed19d7.png)

**Comment by rennokki** [comment]
I had encountered this problem before. It has to be a network problem, and the pods are awaiting the response from the Redis instance, and actively consuming CPU/Memory for the requests.

This can be network throttling (usually, the network is the bottleneck before the actual performance issue with Redis/soketi). Swapping to `cluster` definitely shows up it's not a soketi issue, but rather a Redis issue.

What I recommend is to deploy soketi in the same private network as the Redis instance and use the private hostname/ip to connect to Redis, as usually, for what I have seen, latency for AWS/GCP's connections from outside the VPC are really slow. Tried to deploy a Postgres instance on GKE and it usually took 1.7s to resolve it, which is super yikes.
